frame/staking/src/slashing.rs

```diff
@@ -597,14 +597,9 @@ pub fn do_slash<T: Config>(
 	slashed_imbalance: &mut NegativeImbalanceOf<T>,
 	slash_era: EraIndex,
 ) {
-	let controller = match <Pallet<T>>::bonded(stash).defensive() {
-		None => return,
-		Some(c) => c,
-	};
-
-	let mut ledger = match <Pallet<T>>::ledger(&controller) {
-		Some(ledger) => ledger,
-		None => return, // nothing to do.
+	let mut ledger = match Pallet::<T>::ledger(sp_staking::StakingAccount::Stash(stash.clone())) {
```

Comment:
You dropped the defensive part about this leading to `Err` it seems?

Reply:
Good spot, added the defensive back.